mentorshipGame/Assets/Scripts/AllyLogic.cs

```diff
@@ -0,0 +1,405 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+//using WeaponSupervisor;
+
+public class AllyLogic : MonoBehaviour
+{
+    //bool PlayerAttack = false; 
+    //bool SpAvailable = false;
+    public WeaponAttackController weaponAttack;
+    public bool partyLeader;
```

Comment:
this should be controlling the party lead.  currently it does nothing.  stop using the function.